Automattic/wp-calypso
# Add `lib/track-element-size`.

The new library provides React hooks to keep track of an element's size in the DOM.
This PR also rewrites the Chart component to use the new library.

This PR doesn't introduce any performance fixes per se, but rather adds building blocks that encapsulate all of the complexity of tracking a component's size in a performant way, making it easier to build performant new components in the future.

#### Changes proposed in this Pull Request

* Add `lib/track-element-size`
* Rewrite `Chart` to make use of `lib/track-element-size`

#### Testing instructions

* Ensure the unit tests for `lib/track-element-size` pass.
* Ensure that the Chart (in Stats) continues to work normally.
